src/hooks/useUploadsManager/useUploadsManager.tsx

```diff
@@ -28,29 +22,30 @@ type GroupByParentObjectIdAcc = {
   [key: string]: AssetUploadWithProgress[]
 }
 
-type AssetFile = {
-  contentId: string
-  blob: File | Blob
-}
-
 const UploadManagerContext = React.createContext<UploadManagerValue | undefined>(undefined)
 UploadManagerContext.displayName = 'UploadManagerContext'
 
 export const UploadManagerProvider: React.FC = ({ children }) => {
   const navigate = useNavigate()
-  const { uploadsState, addAsset, updateAsset } = useUploadsManagerStore()
   const { getStorageProvider, markStorageProviderNotWorking } = useStorageProviders()
-  const { displaySnackbar } = useSnackbar()
-  const [uploadsProgress, setUploadsProgress] = useState<UploadsProgressRecord>({})
+
+  const displaySnackbar = useStore((state) => state.displaySnackbar)
+  const addAsset = useStore((state) => state.addAsset)
+  const updateAsset = useStore((state) => state.updateAsset)
+  const uploadsState = useStore((state) => state.uploadsState)
+  const uploadsProgress = useStore((state) => state.uploadsProgress)
+  const setUploadsProgress = useStore((state) => state.setUploadsProgress)
+  const assetsFiles = useStore((state) => state.assetsFiles)
+  const setAssetsFiles = useStore((state) => state.setAssetsFiles)
```

Comment:
Possibly I could do: 
```tsx
const {displaySnackbar, addAsset, updateAsset, uploadsState, uploadsProgress, setUploadsProgress, assetsFiles, setAssetsFiles} = useStore()
``` 
but I'm afraid that it may cause some unintended rerenders - not 100% sure

Ideally, as I mentioned [before](https://github.com/Joystream/atlas/pull/742/files#r649792760) - we should just split the global store into smaller stores.